// pipedrive/dist/model/InlineResponse4031AdditionalData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The InlineResponse4031AdditionalData model module.
 * @module model/InlineResponse4031AdditionalData
 * @version 1.0.0
 */
var InlineResponse4031AdditionalData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InlineResponse4031AdditionalData</code>.
   * @alias module:model/InlineResponse4031AdditionalData
   */
  function InlineResponse4031AdditionalData() {
    (0, _classCallCheck2["default"])(this, InlineResponse4031AdditionalData);
    InlineResponse4031AdditionalData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(InlineResponse4031AdditionalData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>InlineResponse4031AdditionalData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InlineResponse4031AdditionalData} obj Optional instance to populate.
     * @return {module:model/InlineResponse4031AdditionalData} The populated <code>InlineResponse4031AdditionalData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InlineResponse4031AdditionalData();

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
          delete data['code'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return InlineResponse4031AdditionalData;
}();
/**
 * An error code sent by the API
 * @member {String} code
 */


InlineResponse4031AdditionalData.prototype['code'] = undefined;
var _default = InlineResponse4031AdditionalData;
exports["default"] = _default;
