// pipedrive/dist/model/InlineResponse403.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _InlineResponse403AdditionalData = _interopRequireDefault(require("./InlineResponse403AdditionalData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The InlineResponse403 model module.
 * @module model/InlineResponse403
 * @version 1.0.0
 */
var InlineResponse403 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InlineResponse403</code>.
   * @alias module:model/InlineResponse403
   */
  function InlineResponse403() {
    (0, _classCallCheck2["default"])(this, InlineResponse403);
    InlineResponse403.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(InlineResponse403, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>InlineResponse403</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InlineResponse403} obj Optional instance to populate.
     * @return {module:model/InlineResponse403} The populated <code>InlineResponse403</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InlineResponse403();

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('error')) {
          obj['error'] = _ApiClient["default"].convertToType(data['error'], 'String');
          delete data['error'];
        }

        if (data.hasOwnProperty('error_info')) {
          obj['error_info'] = _ApiClient["default"].convertToType(data['error_info'], 'String');
          delete data['error_info'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _InlineResponse403AdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return InlineResponse403;
}();
/**
 * @member {Boolean} success
 */


InlineResponse403.prototype['success'] = undefined;
/**
 * The error description
 * @member {String} error
 */

InlineResponse403.prototype['error'] = undefined;
/**
 * @member {String} error_info
 */

InlineResponse403.prototype['error_info'] = undefined;
/**
 * @member {module:model/InlineResponse403AdditionalData} additional_data
 */

InlineResponse403.prototype['additional_data'] = undefined;
var _default = InlineResponse403;
exports["default"] = _default;
