// pipedrive/dist/model/LeadSearchResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _ItemSearchAdditionalData = _interopRequireDefault(require("./ItemSearchAdditionalData"));

var _LeadSearchResponseAllOf = _interopRequireDefault(require("./LeadSearchResponseAllOf"));

var _LeadSearchResponseAllOfData = _interopRequireDefault(require("./LeadSearchResponseAllOfData"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The LeadSearchResponse model module.
 * @module model/LeadSearchResponse
 * @version 1.0.0
 */
var LeadSearchResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LeadSearchResponse</code>.
   * @alias module:model/LeadSearchResponse
   * @implements module:model/BaseResponse
   * @implements module:model/LeadSearchResponseAllOf
   */
  function LeadSearchResponse() {
    (0, _classCallCheck2["default"])(this, LeadSearchResponse);

    _BaseResponse["default"].initialize(this);

    _LeadSearchResponseAllOf["default"].initialize(this);

    LeadSearchResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(LeadSearchResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>LeadSearchResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LeadSearchResponse} obj Optional instance to populate.
     * @return {module:model/LeadSearchResponse} The populated <code>LeadSearchResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LeadSearchResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _LeadSearchResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _LeadSearchResponseAllOfData["default"].constructFromObject(data['data']);
          delete data['data'];
        }

        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _ItemSearchAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return LeadSearchResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


LeadSearchResponse.prototype['success'] = undefined;
/**
 * @member {module:model/LeadSearchResponseAllOfData} data
 */

LeadSearchResponse.prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

LeadSearchResponse.prototype['additional_data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement LeadSearchResponseAllOf interface:

/**
 * @member {module:model/LeadSearchResponseAllOfData} data
 */

_LeadSearchResponseAllOf["default"].prototype['data'] = undefined;
/**
 * @member {module:model/ItemSearchAdditionalData} additional_data
 */

_LeadSearchResponseAllOf["default"].prototype['additional_data'] = undefined;
var _default = LeadSearchResponse;
exports["default"] = _default;
