// pipedrive/dist/model/LeadSearchItemItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _LeadSearchItemItemOrganization = _interopRequireDefault(require("./LeadSearchItemItemOrganization"));

var _LeadSearchItemItemOwner = _interopRequireDefault(require("./LeadSearchItemItemOwner"));

var _LeadSearchItemItemPerson = _interopRequireDefault(require("./LeadSearchItemItemPerson"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The LeadSearchItemItem model module.
 * @module model/LeadSearchItemItem
 * @version 1.0.0
 */
var LeadSearchItemItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LeadSearchItemItem</code>.
   * @alias module:model/LeadSearchItemItem
   */
  function LeadSearchItemItem() {
    (0, _classCallCheck2["default"])(this, LeadSearchItemItem);
    LeadSearchItemItem.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(LeadSearchItemItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>LeadSearchItemItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LeadSearchItemItem} obj Optional instance to populate.
     * @return {module:model/LeadSearchItemItem} The populated <code>LeadSearchItemItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LeadSearchItemItem();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
          delete data['id'];
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
          delete data['type'];
        }

        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
          delete data['title'];
        }

        if (data.hasOwnProperty('owner')) {
          obj['owner'] = _LeadSearchItemItemOwner["default"].constructFromObject(data['owner']);
          delete data['owner'];
        }

        if (data.hasOwnProperty('person')) {
          obj['person'] = _LeadSearchItemItemPerson["default"].constructFromObject(data['person']);
          delete data['person'];
        }

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _LeadSearchItemItemOrganization["default"].constructFromObject(data['organization']);
          delete data['organization'];
        }

        if (data.hasOwnProperty('phones')) {
          obj['phones'] = _ApiClient["default"].convertToType(data['phones'], ['String']);
          delete data['phones'];
        }

        if (data.hasOwnProperty('emails')) {
          obj['emails'] = _ApiClient["default"].convertToType(data['emails'], ['String']);
          delete data['emails'];
        }

        if (data.hasOwnProperty('custom_fields')) {
          obj['custom_fields'] = _ApiClient["default"].convertToType(data['custom_fields'], 'Number');
          delete data['custom_fields'];
        }

        if (data.hasOwnProperty('notes')) {
          obj['notes'] = _ApiClient["default"].convertToType(data['notes'], ['String']);
          delete data['notes'];
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'Number');
          delete data['value'];
        }

        if (data.hasOwnProperty('currency')) {
          obj['currency'] = _ApiClient["default"].convertToType(data['currency'], 'String');
          delete data['currency'];
        }

        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], 'Number');
          delete data['visible_to'];
        }

        if (data.hasOwnProperty('is_archived')) {
          obj['is_archived'] = _ApiClient["default"].convertToType(data['is_archived'], 'Boolean');
          delete data['is_archived'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return LeadSearchItemItem;
}();
/**
 * The ID of the lead
 * @member {String} id
 */


LeadSearchItemItem.prototype['id'] = undefined;
/**
 * The type of the item
 * @member {String} type
 */

LeadSearchItemItem.prototype['type'] = undefined;
/**
 * The title of the lead
 * @member {String} title
 */

LeadSearchItemItem.prototype['title'] = undefined;
/**
 * @member {module:model/LeadSearchItemItemOwner} owner
 */

LeadSearchItemItem.prototype['owner'] = undefined;
/**
 * @member {module:model/LeadSearchItemItemPerson} person
 */

LeadSearchItemItem.prototype['person'] = undefined;
/**
 * @member {module:model/LeadSearchItemItemOrganization} organization
 */

LeadSearchItemItem.prototype['organization'] = undefined;
/**
 * @member {Array.<String>} phones
 */

LeadSearchItemItem.prototype['phones'] = undefined;
/**
 * @member {Array.<String>} emails
 */

LeadSearchItemItem.prototype['emails'] = undefined;
/**
 * Custom fields
 * @member {Number} custom_fields
 */

LeadSearchItemItem.prototype['custom_fields'] = undefined;
/**
 * An array of notes
 * @member {Array.<String>} notes
 */

LeadSearchItemItem.prototype['notes'] = undefined;
/**
 * The value of the lead
 * @member {Number} value
 */

LeadSearchItemItem.prototype['value'] = undefined;
/**
 * The currency of the lead
 * @member {String} currency
 */

LeadSearchItemItem.prototype['currency'] = undefined;
/**
 * The visibility of the lead
 * @member {Number} visible_to
 */

LeadSearchItemItem.prototype['visible_to'] = undefined;
/**
 * A flag indicating whether the lead is archived or not
 * @member {Boolean} is_archived
 */

LeadSearchItemItem.prototype['is_archived'] = undefined;
var _default = LeadSearchItemItem;
exports["default"] = _default;
