// pipedrive/dist/model/LeadSearchItemItemPerson.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The LeadSearchItemItemPerson model module.
 * @module model/LeadSearchItemItemPerson
 * @version 1.0.0
 */
var LeadSearchItemItemPerson = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LeadSearchItemItemPerson</code>.
   * @alias module:model/LeadSearchItemItemPerson
   */
  function LeadSearchItemItemPerson() {
    (0, _classCallCheck2["default"])(this, LeadSearchItemItemPerson);
    LeadSearchItemItemPerson.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(LeadSearchItemItemPerson, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>LeadSearchItemItemPerson</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LeadSearchItemItemPerson} obj Optional instance to populate.
     * @return {module:model/LeadSearchItemItemPerson} The populated <code>LeadSearchItemItemPerson</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LeadSearchItemItemPerson();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return LeadSearchItemItemPerson;
}();
/**
 * The ID of the person the lead is associated with
 * @member {Number} id
 */


LeadSearchItemItemPerson.prototype['id'] = undefined;
/**
 * The name of the person the lead is associated with
 * @member {String} name
 */

LeadSearchItemItemPerson.prototype['name'] = undefined;
var _default = LeadSearchItemItemPerson;
exports["default"] = _default;
