// pipedrive/dist/model/NewPersonAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The NewPersonAllOf model module.
 * @module model/NewPersonAllOf
 * @version 1.0.0
 */
var NewPersonAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NewPersonAllOf</code>.
   * @alias module:model/NewPersonAllOf
   * @param name {String} The name of the person
   */
  function NewPersonAllOf(name) {
    (0, _classCallCheck2["default"])(this, NewPersonAllOf);
    NewPersonAllOf.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(NewPersonAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>NewPersonAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NewPersonAllOf} obj Optional instance to populate.
     * @return {module:model/NewPersonAllOf} The populated <code>NewPersonAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NewPersonAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return NewPersonAllOf;
}();
/**
 * The name of the person
 * @member {String} name
 */


NewPersonAllOf.prototype['name'] = undefined;
var _default = NewPersonAllOf;
exports["default"] = _default;
