// pipedrive/dist/model/SubscriptionAddonsResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BaseResponse = _interopRequireDefault(require("./BaseResponse"));

var _SubscriptionAddonsResponseAllOf = _interopRequireDefault(require("./SubscriptionAddonsResponseAllOf"));

/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

/**
 * The SubscriptionAddonsResponse model module.
 * @module model/SubscriptionAddonsResponse
 * @version 1.0.0
 */
var SubscriptionAddonsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SubscriptionAddonsResponse</code>.
   * @alias module:model/SubscriptionAddonsResponse
   * @implements module:model/BaseResponse
   * @implements module:model/SubscriptionAddonsResponseAllOf
   */
  function SubscriptionAddonsResponse() {
    (0, _classCallCheck2["default"])(this, SubscriptionAddonsResponse);

    _BaseResponse["default"].initialize(this);

    _SubscriptionAddonsResponseAllOf["default"].initialize(this);

    SubscriptionAddonsResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  (0, _createClass2["default"])(SubscriptionAddonsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SubscriptionAddonsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SubscriptionAddonsResponse} obj Optional instance to populate.
     * @return {module:model/SubscriptionAddonsResponse} The populated <code>SubscriptionAddonsResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SubscriptionAddonsResponse();

        _BaseResponse["default"].constructFromObject(data, obj);

        _SubscriptionAddonsResponseAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [Object]);
          delete data['data'];
        }

        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }

      return obj;
    }
  }]);
  return SubscriptionAddonsResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */


SubscriptionAddonsResponse.prototype['success'] = undefined;
/**
 * An array of add-ons that the company has.
 * @member {Array.<Object>} data
 */

SubscriptionAddonsResponse.prototype['data'] = undefined; // Implement BaseResponse interface:

/**
 * If the response is successful or not
 * @member {Boolean} success
 */

_BaseResponse["default"].prototype['success'] = undefined; // Implement SubscriptionAddonsResponseAllOf interface:

/**
 * An array of add-ons that the company has.
 * @member {Array.<Object>} data
 */

_SubscriptionAddonsResponseAllOf["default"].prototype['data'] = undefined;
var _default = SubscriptionAddonsResponse;
exports["default"] = _default;
